e2e/key/keyprivate/keyprivate.go

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package keyprivate
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/sylabs/singularity/e2e/internal/keyexec"
+	"github.com/sylabs/singularity/internal/pkg/test"
+)
+
+type testingEnv struct {
+	// base env for running tests
+	CmdPath     string `split_words:"true"`
+	TestDir     string `split_words:"true"`
+	RunDisabled bool   `default:"false"`
+}
+
+var testenv testingEnv
+
+var keyPath string
+var defaultKeyFile string
+
+func testPrivateKeyNewPair(t *testing.T) {
+	tests := []struct {
+		name    string
+		user    string
+		email   string
+		note    string
+		psk1    string
+		psk2    string
+		push    bool
+		succeed bool
+	}{
+		{
+			name:    "newpair",
+			user:    "genbye2etests\n",
+			email:   "westley@sylabs.io\n",
+			note:    "test key generated by e2e tests\n",
+			psk1:    "e2etests\n",
```

Comment:
psk2 is not set, is that intentional?

Reply:
Yes, if `psk2` is not set, `keyexec.KeyNewPair()` will use the same password as `psk1`.

Reply:
uhg.

:-)

Reply:
Okay, I'll remove `psk2`, its not necessary right now.